src/NUnitFramework/tests/Internal/Filters/PartitionFilterTests.cs

```diff
@@ -106,5 +106,24 @@ public void ComputeParitionNumberHandlesLongTestCaseName()
             Assert.DoesNotThrow(() => _filter.ComputePartitionNumber(_testFixtureWithWithLongTestCaseNames.Tests[0]));
             Assert.DoesNotThrow(() => _filter.ComputePartitionNumber(_testFixtureWithWithLongTestCaseNames.Tests[1]));
         }
+
+        [TestCase("1/1n")]
+        [TestCase("1")]
+        public void TryCreateFailure(string input)
+        {
+            Assert.That(PartitionFilter.TryCreate(input, out var filter), Is.False);
+        }
+
+        [TestCase("1/2")]
+        [TestCase(" 1/2")]
+        [TestCase("1/2 ")]
+        public void TryCreateSuccess(string input)
+        {
+            var result = PartitionFilter.TryCreate(input, out var filter);
+
+            Assert.That(result, Is.True);
+            Assert.That(filter!.PartitionNumber, Is.EqualTo(1));
```

Comment:
question for the reviewer: I added `NotNullWhen` on the `TryCreate` method in the hopes that I wouldn't need to use the `!` operator here. I had thought the analyzer would already detect a situation like this where we reference it _after_ asserting that `result Is.True`. Am I misremembering or mis-decorating?